spec/libraries/helpers_spec.rb

```diff
@@ -0,0 +1,195 @@
+require 'spec_helper'
+
+RSpec.describe Pdns::Helpers do
+  subject do
+    class DummyClass < Chef::Node
+      include Pdns::Helpers
+    end
+    DummyClass.new
+  end
+
+  describe '#default_user_attributes' do
+    context 'when platform is debian' do
+      it 'returns a Mash for the Debian system user and home path' do
+        allow(subject).to receive(:[]).with('platform_family').and_return('debian')
+        expect(subject.default_user_attributes).to eq Mash.new(home: '/var/spool/powerdns', shell: '/bin/false')
+      end
+    end
+
+    context 'when platform is rhel' do
+      it 'returns a Mash for the RHEL system user and home path' do
+        allow(subject).to receive(:[]).with('platform_family').and_return('rhel')
+        expect(subject.default_user_attributes).to eq Mash.new(home: '/', shell: '/sbin/nologin')
+      end
+    end
+  end
+end
+
+RSpec.describe Pdns::RecursorHelpers do
+  subject do
+    class DummyClass < Chef::Node
+      include Pdns::RecursorHelpers
+    end
+    DummyClass.new
+  end
+
+  describe '#systemd_name' do
+    context 'without a name' do
+      let(:instance) { '' }
+      it 'returns the service name without a specific name' do
+        expect(subject.systemd_name(instance)).to eq 'pdns-recursor'
+      end
+    end
+
+    context 'with a name' do
+      let(:instance) { 'foo' }
+      it 'returns the service name with a virtual instance name' do
+        expect(subject.systemd_name(instance)).to eq('pdns-recursor@foo')
+      end
+    end
+  end
+
+  describe '#sysvinit_name' do
+    context 'without a name' do
+      let(:instance) { '' }
+      it 'returns the service name without a specific name' do
+        expect(subject.sysvinit_name(instance)).to eq 'pdns-recursor'
+      end
+    end
+
+    context 'with a name' do
+      let(:instance) { 'foo' }
+      it 'returns the service name with a virtual instance name' do
+        expect(subject.sysvinit_name(instance)).to eq('pdns-recursor-foo')
```

Comment:
It should not. The name should include the hyphen there. The way PowerDNS does naming in their docs states you should have a hyphen between the instance name and the init script, but don't add a hyphen in the name itself because it splits the name based upon the hyphen in the script name.